controller.rb

```diff
@@ -15,4 +15,10 @@ def rescue
     animal = Animal.new(type, name)
     @register.add(animal)
   end
-end
\ No newline at end of file
+
+  def adopt
+    @view.list_all_animals(@register.all)
+    index = @view.ask_for_animal_index
+    @view.adopted_animal(@register.adopt(index).name)
```

Comment:
I think it would be a bit easier to read if you split this up into separate lines.
```
adopted_animal = @register.adopt(index)
@view.adopted_animal(adopted_animal.name)
```